### PR TITLE
persepolis: init at 3.1.0

### PR DESCRIPTION
**Pull request by LinArcX**
###### Motivation for this change
It is one the best gui download manager in linux world. :)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/networking/persepolis/default.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+     mkdir -p $out/share/applications
+     cp ${desktopItem}/share/applications/* $out/share/applications
+  '';
+
+  # see: https://github.com/persepolisdm/persepolis/blob/3.1.0/setup.py#L130
+  doCheck = false;
+
+  propagatedBuildInputs = [
+    aria
+    libnotify
```
Comment (aanderse) [inline pkgs/tools/networking/persepolis/default.nix]: After reading [setup.py](https://github.com/persepolisdm/persepolis/blob/master/setup.py) I'm wondering if `aria` and possibly `libnotify` don't belong in propagatedBuildInputs, but instead if the program should use `makeWrapper` to include `aria` and `libnotify` in the PATH?

Comment (LinArcX) [reply pkgs/tools/networking/persepolis/default.nix]: Something like this:
https://gist.github.com/LinArcX/d1d9742806fcc12624eedfcb5866a7d4

Comment (aanderse) [reply pkgs/tools/networking/persepolis/default.nix]: Yes, but also including `wrapProgram $out/bin/persepolis --prefix PATH : "${lib.makeBinPath [aria libnotify ]}"`, assuming `persepolis` is the only binary.

Comment (LinArcX) [reply pkgs/tools/networking/persepolis/default.nix]: Done!

Comment (aanderse) [reply pkgs/tools/networking/persepolis/default.nix]: Awesome!

**Comment by aanderse** [comment]
@LinArcX I built your branch and ran it. I opened and closed the help dialog (via the window managers close button) which caused the application to crash. Same thing for preferences. Did that happen for you as well?

**Comment by LinArcX** [comment]
@aanderse No. After last change I test it locally. It's work like a charm now.
Also I've downloaded some musics with it to test its functionality. It's OK.

**Review by aanderse — APPROVED** [review]
Played around a bit more and it only crashes when I use the window manager to close... so I'll assume that is a window manager issue? LGTM :+1:

**Comment by LinArcX** [comment]
Nice. Also I've tested it in i3. Without any problems. 
So what is next step? :)

File: pkgs/tools/networking/persepolis/default.nix [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    owner = "persepolisdm";
+    repo = "persepolis";
+    rev = "${version}";
+    sha256 = "0xngk8wgj5k27mh3bcrf2wwzqr8a3g0d4pc5i5vcavnnaj03j44m";
+  };
+
+  # see: https://github.com/persepolisdm/persepolis/blob/3.1.0/setup.py#L130
+  doCheck = false;
+
+  desktopItem = makeDesktopItem {
```
Comment (jtojnar) [inline pkgs/tools/networking/persepolis/default.nix]: There is a better desktop file in `xdg` directory already:

https://github.com/persepolisdm/persepolis/blob/master/xdg/com.github.persepolisdm.persepolis.desktop

Comment (LinArcX) [reply pkgs/tools/networking/persepolis/default.nix]: Done!

**Pull request merged**